packages/instant-apps-components/src/components/instant-apps-language-switcher/instant-apps-language-switcher.tsx

```diff
@@ -0,0 +1,175 @@
+    locale: string;
+    data?: {
+      [key: string]: string;
+    };
+  }>;
+
+  async componentWillLoad() {
+    const [intl] = await loadModules(['esri/intl']);
+    this.intl = intl;
+    this.messages = await getMessages(document.createElement('instant-apps-language-translator'));
+    this.portalItemResourceT9n = await this.getPortalItemResourceT9n();
+    const t9nData = await getPortalItemResourceT9nData(this.portalItemResourceT9n);
+    const lang = getComponentClosestLanguage(document.body) as string;
+    if (lang) {
+      this.userLocale = lang;
+      this.selectedLanguage = lang;
+    }
+    if (t9nData) {
+      this.t9nData = t9nData;
+      this.locales = Object.keys(t9nData);
+    }
+  }
+
+  render() {
+    const trigger = this.renderTrigger();
+    const dropdown = this.renderDropdownItems();
+    const { userLocale } = this;
+    return (
+      <calcite-dropdown width="m">
+        {trigger}
+        <calcite-dropdown-item
+          key={`default-${userLocale}`}
+          value={userLocale}
+          onCalciteDropdownItemSelect={this.calciteDropdownItemSelectCallback(userLocale)}
+          selected={this.selectedLanguage === this.userLocale}
+        >
+          {this.getSelectedLanguageText(userLocale)}
+        </calcite-dropdown-item>
+        {dropdown}
+      </calcite-dropdown>
+    );
+  }
+
+  renderTrigger(): HTMLCalciteActionElement {
+    return <calcite-action slot="trigger" icon={this.icon} text={this.getSelectedLanguageText(this.selectedLanguage as string)} text-enabled={true} />;
+  }
+
+  renderDropdownItems(): HTMLCalciteDropdownItemElement[] {
+    return this.locales?.map(translatedLanguage => this.renderDropdownItem(translatedLanguage));
+  }
+
+  renderDropdownItem(translatedLanguage: string): HTMLCalciteDropdownItemElement {
+    const text = this.getSelectedLanguageText(translatedLanguage);
+    const selected = translatedLanguage === this.selectedLanguage;
+
+    return (
+      <calcite-dropdown-item
+        key={translatedLanguage}
+        value={translatedLanguage}
+        selected={selected}
+        onCalciteDropdownItemSelect={this.calciteDropdownItemSelectCallback(translatedLanguage)}
+      >
+        {text}
+      </calcite-dropdown-item>
+    );
+  }
+
+  calciteDropdownItemSelectCallback(translatedLanguage: string): () => void {
+    return () => {
+      this.selectedLanguage = translatedLanguage;
+      const eventData = {
+        locale: this.selectedLanguage,
+      };
+
+      if (this.selectedLanguage !== this.userLocale) {
+        eventData['data'] = this.t9nData[translatedLanguage];
+      }
+      this.intl.setLocale(this.selectedLanguage);
+
+      // Set url parameter 'locale' with value
+      const params = new URLSearchParams(window.location.search);
+      params.set('locale', this.selectedLanguage);
+      document.documentElement.setAttribute('lang', this.selectedLanguage);
+
+      if (this.selectedLanguage === 'ar' || this.selectedLanguage === 'he') {
```

Comment:
This is fine if you aren't loading the api but if you have to do it for this project might be worth using the esri/intl prefersRTL instead? https://developers.arcgis.com/javascript/latest/api-reference/esri-intl.html#prefersRTL

Reply:
Done